File: Data Science and Machine Learning Essentials/Lab3A Exploring and Visualizing Data/DAT203x_LAB03A/PrepEE.R

## Load the data
dirName <- 'C:\\DAT203xLabfiles'
fileName <- "EnergyEfficiencyRegressiondata.csv"
infile <- file.path(dirName, fileName)
eeframe <- read.csv( infile, header = TRUE, stringsAsFactors = FALSE)

## Remove dots from column names.
names(eeframe) <- gsub("\\.", "", names(eeframe))

## Remove columns we are not going to use.
eeframe$CoolingLoad <- NULL

## Convert some columns to factors/categorical. 
catList <- c("OverallHeight", "Orientation")
eeframe[, catList] <- lapply(eeframe[, catList], 
                             function(x) as.factor(as.character(x)))

## Scale the numeric features. 
scaleList <- c("RelativeCompactness", "SurfaceArea",
               "WallArea", "RoofArea", "GlazingArea", 
               "GlazingAreaDistribution")
eeframe[, scaleList] <- lapply(eeframe[, scaleList], function(x) as.numeric(scale(x)))

File: Data Science and Machine Learning Essentials/Lab3A Exploring and Visualizing Data/DAT203x_LAB03A/PrepEE.py

## Load the data  
import pandas as pd
import os
from sklearn import preprocessing

pathName = "c://DAT203xLabfiles"
fileName = "EnergyEfficiencyRegressiondata.csv"
filePath = os.path.join(pathName, fileName)
eeframe = pd.read_csv(filePath)

## Remove columns we're not going to use
eeframe = eeframe.drop('Cooling Load', 1)  

## scale numeric features
scaleList = ["Relative Compactness", "Surface Area", \
       "Wall Area", "Roof Area", "Glazing Area",  \
       "Glazing Area Distribution"]
arry = eeframe[scaleList].as_matrix()
eeframe[scaleList] = preprocessing.scale(arry)  
        
